app/src/main/res/layout/random_preset_fragment.xml

```diff
@@ -0,0 +1,125 @@
+      android:textAppearance="?attr/textAppearanceHeadline5"
+      android:layout_marginTop="10dp"/>
+
+    <RadioGroup
+      android:id="@+id/preset_type"
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:checkedButton="@id/preset_type__any"
+      android:layout_marginStart="10dp">
+
+      <RadioButton
+        android:id="@+id/preset_type__any"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/any" />
+
+      <RadioButton
+        android:id="@+id/preset_type__focus"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/focus" />
+
+      <RadioButton
+        android:id="@+id/preset_type__relax"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/relax" />
+
+    </RadioGroup>
+
+    <TextView
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:layout_marginTop="20dp"
+      android:layout_marginBottom="7dp"
+      android:text="@string/intensity"
+      android:textAppearance="?attr/textAppearanceHeadline5" />
+
+    <TextView
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:layout_marginVertical="7dp"
+      android:text="@string/intensity_description" />
+
+    <RadioGroup
+      android:id="@+id/preset_intensity"
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:checkedButton="@id/preset_intensity__any"
+      android:layout_marginStart="10dp">
+
+      <RadioButton
+        android:id="@+id/preset_intensity__any"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/any" />
+
+      <RadioButton
+        android:id="@+id/preset_intensity__light"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/light" />
+
+      <RadioButton
+        android:id="@+id/preset_intensity__dense"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/dense" />
+
+    </RadioGroup>
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+      android:layout_width="wrap_content"
+      android:layout_height="wrap_content"
+      android:layout_gravity="end">
+
+      <Button
+        android:id="@+id/cancel_preset_button"
+        style="?attr/buttonStyle"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="16dp"
+        android:visibility="visible"
+        android:text="@string/cancel"
+        app:layout_constraintEnd_toStartOf="@id/play_preset_button"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintBottom_toBottomOf="parent" />
```

Comment:
This isn't really needed...? There's nothing to cancel.

Reply:
You're right! I just followed the exact same picture layout, but it was still kinda weird for me looking at the buttons there, it looked pretty ugly, and as you said play button could be a fab! that's why I decided to push these changes before continue doing more work.